pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -84,11 +81,6 @@ public TopicSchema(PulsarClient client) {
     private SchemaType getSchemaTypeOrDefault(String topic, Class<?> clazz) {
         if (GenericRecord.class.isAssignableFrom(clazz)) {
             return SchemaType.AUTO_CONSUME;
-        } else if (byte[].class.equals(clazz)
```

Comment:
Currently this logic seems to be spread out to multiple places. i removed this check and moved it to getDefaultSchemaType().